zkevm-circuits/src/evm_circuit/util/math_gadget/mul_add_words.rs

```diff
@@ -59,15 +63,15 @@ impl<F: Field> MulAddWordsGadget<F> {
 
         let mut a_limbs = vec![];
         let mut b_limbs = vec![];
-        for trunk in 0..4 {
-            let idx = (trunk * 8) as usize;
-            a_limbs.push(from_bytes::expr(&a.cells[idx..idx + 8]));
-            b_limbs.push(from_bytes::expr(&b.cells[idx..idx + 8]));
+        let word4_a: Word4<Expression<F>> = a.expr().to_wordlimbs();
```

Comment:
This looks really nice!  I think having these gadgets take `Word32` and then converting to a different number of limbs as necessary makes the refactor look very clean!